### PR TITLE
migration: Do not crash if database is ahead a version

### PR DESCRIPTION
**Pull request by efritz**
Noticed a behavior change. Not sure if we upgraded golang-migrate, but we weren't catching this wrapped error condition.

Currently if you set your db version ahead (or go back to `main` from a feature branch with a migration) you cannot start you stack. This also means that while this behavior was in effect, customers will not be able to downgrade without first reverting the schema.

This PR ensures that having a higher schema version is a non-fatal condition in frontend startup.

### CODE REVIEW

File: internal/database/dbconn/migration.go [hunk 1]
```diff
@@ -99,10 +99,11 @@ func NewMigrate(db *sql.DB, database *Database) (*migrate.Migrate, error) {
 func DoMigrate(m *migrate.Migrate) (err error) {
 	err = m.Up()
 	if err == nil || err == migrate.ErrNoChange {
+		fmt.Printf("A\n")
```
Comment (efritz) [inline internal/database/dbconn/migration.go:102]:
```suggestion
```

File: internal/database/dbconn/migration.go [hunk 2]
```diff
@@ -111,9 +112,11 @@ func DoMigrate(m *migrate.Migrate) (err error) {
 		if dirty { // this shouldn't happen, but checking anyways
 			return err
 		}
+
 		log15.Warn("WARNING: Detected an old version of Sourcegraph. The database has migrated to a newer version. If you have applied a rollback, this is expected and you can ignore this warning. If not, please contact support@sourcegraph.com for further assistance.", "db_version", version)
 		return nil
 	}
+
```
Comment (efritz) [inline internal/database/dbconn/migration.go:118]:
```suggestion
```

File: internal/database/dbconn/migration.go [hunk 3]
```diff
@@ -111,6 +111,7 @@ func DoMigrate(m *migrate.Migrate) (err error) {
 		if dirty { // this shouldn't happen, but checking anyways
 			return err
 		}
+
```
Comment (efritz) [inline internal/database/dbconn/migration.go:114]:
```suggestion
```

**Review by eseliger — APPROVED** [review]
Nice catch

**Review by mrnugget — APPROVED** [review]

**Pull request merged**